app/prompts/admin.py

```diff
@@ -0,0 +1,333 @@
+    extra = 1
+    ordering = ["value"]
+    fields = [
+        "label",
+        "value",
+    ]
+
+
+class RangedPromptAdmin(SetDisabledByWhenDisabledAdmin):
+    model = RangedPrompt
+    form = RangedPromptForm
+    inlines = [RangedPromptOptionAdmin]
+    extra = 1
+    ordering = ["feedback_form", "order"]
+    fields = [
+        "uuid",
+        "feedback_form",
+        "order",
+        "is_disabled",
+        "disabled",
+        "text",
+        "created_by",
+    ]
+    readonly_fields = [
+        "uuid",
+        "feedback_form",
+        "order",
+        "disabled",
+        "created_by",
+    ]
+    list_display = ["text", "options", "feedback_form", "order", "uuid"]
+    list_filter = ["feedback_form"]
+    search_fields = ["text", "uuid", "options__label"]
+
+    def disabled(self, instance):
+        return "Yes" if instance.disabled_at else "No"
+
+    disabled.short_description = "Disabled"
+
+    def has_add_permission(self, request, obj=None):
+        return False
+
+    def options(self, obj):
+        return ", ".join(
+            option.label for option in obj.options.all().order_by("value")
+        )
+
+    options.short_description = "Path patterns"
+
+    def get_queryset(self, request):
+        query_set = super().get_queryset(request)
+        # Prefetch only changelist page
+        id = request.resolver_match.kwargs.get("object_id")
+        if not id:
+            query_set = query_set.prefetch_related("options")
+
+        return query_set.select_related("created_by", "disabled_by")
+
+
+admin.site.register(TextPrompt, TextPromptAdmin)
+admin.site.register(BinaryPrompt, BinaryPromptAdmin)
+admin.site.register(RangedPrompt, RangedPromptAdmin)
+
+
+class PromptDetailsWidget(forms.Widget):
+    def render(self, name, value, attrs=None, renderer=None):
+        # self.instance is added by PromptForm
+        if not self.instance.id:
+            return ""
+
+        url = reverse(
+            get_prompt_viewname(self.instance),
+            kwargs={"object_id": self.instance.id},
+        )
+        return mark_safe(f'<a href="{url}">Edit details</a>')
+
+
+class PromptForm(IsDisabledCheckboxForm):
+    prompt_details = forms.CharField(
+        widget=PromptDetailsWidget(),
+        label="prompt details",
+        required=False,
+    )
+    prompt_type = forms.ChoiceField(
+        choices=[
+            ("", "Please select..."),
+            *PROMPT_LABELS.items(),
+        ]
+    )
+
+    class Meta:
+        model = Prompt
+        fields = "__all__"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if self.instance:
+            # Pass instance to widget to display link to specific Prompt
+            self.fields["prompt_details"].widget.instance = self.instance
+
+            # For existing prompts set prompt type
+            if self.instance.pk:
+                self.fields["prompt_type"].initial = type(
+                    self.instance
```

Comment:
There's also `instance._meta.model_name` here. I wonder if we ought to prefix with `_meta.app_label` in case future prompts are stored in different apps (likely not).

Reply:
This seems maybe more Django-y but I really hate terms with no word delineation 😡.
Also, I don't see any reason a prefix couldn't be added later, if needed.